crates/ty_python_semantic/src/types/special_form.rs

```diff
@@ -154,9 +154,10 @@ impl SpecialFormType {
             | Self::Intersection
             | Self::CallableTypeOf
             | Self::Protocol  // actually `_ProtocolMeta` at runtime but this is what typeshed says
-            | Self::Generic  // actually `type` at runtime but this is what typeshed says
             | Self::ReadOnly => KnownClass::SpecialForm,
 
+            Self::Generic => KnownClass::Type,
```

Comment:
I do see that a number of our benchmarks (including tomllib) have a 1% regression from this PR, so it's not _just_ the one micro-benchmark (although clearly it's much worse there.)

I suspect the reason for the regression is that there are some cases where we fall back to `KnownClass::Type` instead of to `KnownClass::SpecialForm` now, and that's more expensive?